src/core/print.c

```diff
@@ -0,0 +1,33 @@
+#include <emscripten.h>
+#include <stdio.h>
+
+// Note: We cannot use stdout / stderr const from stdio.h because we override
+// the stream when initializing the Pyodide runtime. Also users can override the
+// streams using `setStderr` and `setStdout`. Therefore, we always need to open
+// the stream on each call.
```

Comment:
I'm confused about this -- don't we refresh `stdout` and `stderr` whenever we change the streams? They are basically just some extra state wrapping file descriptors 1 and 2 and we replace those file descriptors. If this comment is true, I would think it's a bug with our stream handling that should be fixed?

Reply:
But we are removing the `/dev/stdout` and creating it again right when we reset the stdout, isn't it?

the `stdout` macro in `stdio.h` is a [`FILE *` descriptor](https://github.com/openbsd/src/blob/91a4c8b7931217b6b2332ae9b98d4e414ca690dc/include/stdio.h#L198) of `/dev/stdout`, to if /dev/stdout changes it would break the pointer I think.

Reply:
Have you tested it? I'd think the `stdout` `File *` contains a pointer to file descriptor 1. As long as we update file descriptor 1 to the right thing, it should work. `dup2(fd, 1)` can be used to redirect stdout and writes to stdout will follow.

If you can show that this is needed that would be a bug we should fix in our stream handling code rather than work around.

Reply:
Last time when I checked, it failed, but let me check it again.

Reply:
Using `stdout` and `stderr` works. Maybe I made a mistake when I tested before. Thank you for asking again!